packages/docusaurus-plugin-content-docs/src/docs.ts

```diff
@@ -131,6 +131,9 @@ function doProcessDocMetadata({
   } = parseMarkdownString(content);
   const frontMatter = validateDocFrontMatter(unsafeFrontMatter);
 
+  const isDraft =
+    (frontMatter.draft && process.env.NODE_ENV === 'production') || false;
```

Comment:
What about removing `draft: true` docs upfront if it's production, to make sure it never goes into `content`? This way we don't have to filter it out again when generating sidebars.